src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -1957,7 +1981,9 @@ static class C
                 public static void M({{type}}<string> x) { }
             }
             """;
-        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion));
+        var comp = CreateCompilationWithSpanAndMemoryExtensions(source,
+            parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion),
+            noStringToReadOnlySpanConversionInSource: true);
```

Comment:
>noStringToReadOnlySpanConversionInSource: true [](http://example.com/codeflow?start=12&length=46)

Is this the only place `noStringToReadOnlySpanConversionInSource` is used? If so, could we just special-case this use? And is this a scenario where we expect to support the non-standard definition? If not, should we test that we're *not* supporting it?

Reply:
> Is this the only place `noStringToReadOnlySpanConversionInSource` is used? If so, could we just special-case this use?

Yes, you're right, I will add only what's needed from the span source into the test and revert the complicated workaround, thanks.

> And is this a scenario where we expect to support the non-standard definition?

Yes, the non-standard user-defined operator from `string` to `ROS<T>` is expected to be supported here - because we do not ignore UDCs from `null` to `ROS<T>`, so we use the UDC.